addons/base/tests/views.py

```diff
@@ -54,6 +62,23 @@ def test_oauth_finish(self):
         name, args, kwargs = mock_callback.mock_calls[0]
         assert_equal(kwargs['user']._id, self.user._id)
 
+    @mock.patch('website.oauth.views.requests.get')
+    def test_oauth_finish_enable_gv(self, mock_requests_get):
+        url = web_url_for(
+            'oauth_callback',
+            service_name=self.ADDON_SHORT_NAME
+        )
+        query_params = {
+            'code': 'somecode',
+            'state': 'somestatetoken',
+        }
+        with override_flag(ENABLE_GV, active=True):
+            request_url = urlunparse(urlparse(url)._replace(query=urlencode(query_params)))
+            res = self.app.get(request_url, auth=self.user.auth)
+        mock_requests_get.assert_called_with(
+            urlunparse(urlparse(GRAVYVALET_URL)._replace(path='callback', query=urlencode(query_params)))
```

Comment:
note: the full path to the endpoint should be `/v1/oauth/callback` (per the latest updates to https://github.com/CenterForOpenScience/osf.io/pull/10591)

Reply:
super nit: Also, feels a bit weird to construct this the exact same way it's constructed in the code.

A different option might be something like:
```
gv_callback_url = mock_requests_get.call_args[0][0]
parsed_callback_url = urlparse(gv_callback_url)
assert parsed_callback_url.netloc == urlparse(GRAVYVALET_URL).netloc
assert parsed_callback_url.path == `v1/oauth/callback`
assert parse_qs(parsed_callback_url.query) == query_params
```

But this is highly inconsequential, don't worry about it unless this also makes more sense to you.